stm/stm.ml

```diff
@@ -2801,13 +2801,21 @@ let handle_failure (e, info) vcs =
   VCS.print ();
   Exninfo.iraise (e, info)
 
-let snapshot_vio ~doc ~output_native_objects ldir long_f_dot_vo =
+let snapshot_vio ~create_vos ~doc ~output_native_objects ldir long_f_dot_vo =
   let doc = finish ~doc in
   if List.length (VCS.branches ()) > 1 then
     CErrors.user_err ~hdr:"stm" (str"Cannot dump a vio with open proofs");
-  Library.save_library_to ~todo:(dump_snapshot ()) ~output_native_objects
-    ldir long_f_dot_vo
-    (Global.opaque_tables ());
+  (* LATER: when create_vos is true, it could be more efficient to not allocate the futures; but for now it seems useful for synchronization of the workers,
+  below, [snapshot] gets computed even if [create_vos] is true. *)
+  let (tasks,counters) = dump_snapshot() in
+  let except = List.fold_left (fun e (r,_) ->
+     Future.UUIDSet.add r.Stateid.uuid e) Future.UUIDSet.empty tasks in
+  let todo_proofs =
+    if create_vos
+      then Library.ProofsTodoSomeEmpty except
```

Comment:
please rename this constructor, it sounds like `truefalse`. I don't get what except stands for, especially in the second case.

Reply:
This is probably a question for @charguer.

Reply:
The variable name "except" has the same meaning as in the original code from `library.ml`, function `save_library_to`, and `kernel.ml` function `dump`. I figured it was better to reuse the same variable name for the same contents. If you want to rename it to something else, should do so throughout the codebase.